### PR TITLE
A tool that returns system information.

### PR DESCRIPTION
**Pull request by sray7965**
This tool returns information related to CPU, memory, cache, hypervisor, virtualization,
hostname, IP addresses, etc of the system it is running on.

@krobertson @alextoombs - Please review. Just to give you guys a little bit of background I have changed the app so that it can be deployed as an app on containers and can be invoked via http. And even if some information like cache etc cannot be retrieved from the system still the app progresses to get whatever information it can and just displays an error string for the data it could not get or may be a blank line next to the field. These new changes have been made as per the business requirement after my discussion with @juanman2.

### CODE REVIEW

File: example-go-system-info/system_info.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+func main() {
+	http.HandleFunc("/", SystemInfo)
+	err := http.ListenAndServe(":"+os.Getenv("PORT"), nil)
+	if err != nil {
+		log.Fatal("ListenAndServe: ", err)
+	}
+}
+
+func SystemInfo(w http.ResponseWriter, req *http.Request) {
```
Comment (juanman2) [inline example-go-system-info/system_info.go]: godoc please

File: example-go-system-info/system_info.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+	io.WriteString(w, fmt.Sprintf("<li>Target Architecture = %s</li>", otherInfoMap["target_architecture"]))
+	io.WriteString(w, fmt.Sprintf("<li>Target Operating System = %s</li>", otherInfoMap["target_os"]))
+	io.WriteString(w, "</ul>")
+	io.WriteString(w, "</div>")
+
+	io.WriteString(w, "</body>")
+	io.WriteString(w, "</html>")
+}
+
+func cpuInfo() map[string]string {
```
Comment (juanman2) [inline example-go-system-info/system_info.go]: make public, add godoc

File: example-go-system-info/system_info.go [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+	if err != nil {
+		cpuMap["processorModel"] = "Error: The processor model for the current system could not be retrieved."
+	} else {
+		cpuMap["processorModel"] = string(outputProcessorModel)
+	}
+
+	return cpuMap
+}
+
+func memoryInfo() map[string]string {
```
Comment (juanman2) [inline example-go-system-info/system_info.go]: public + godoc

**Comment by juanman2** [comment]
LGTM

File: example-go-system-info/system_info.go (hunk 3 above)
Comment (sray7965) [inline example-go-system-info/system_info.go]: Fixed.

File: example-go-system-info/system_info.go (hunk 2 above)
Comment (sray7965) [inline example-go-system-info/system_info.go]: Fixed.

File: example-go-system-info/system_info.go (hunk 1 above)
Comment (sray7965) [inline example-go-system-info/system_info.go]: Fixed.

**Pull request merged**